pkg/mcs/scheduling/server/cluster.go

```diff
@@ -29,10 +35,94 @@ func NewCluster(ctx context.Context, storage endpoint.RuleStorage, cfg *config.C
 	if err != nil {
 		return nil, err
 	}
+
 	return &Cluster{
-		basicCluster:   basicCluster,
+		BasicCluster:   basicCluster,
 		ruleManager:    placement.NewRuleManager(storage, basicCluster, persistConfig),
 		labelerManager: labelerManager,
 		persistConfig:  persistConfig,
+		hotStat:        statistics.NewHotStat(ctx),
 	}, nil
 }
+
+// GetBasicCluster returns the basic cluster.
+func (c *Cluster) GetBasicCluster() *core.BasicCluster {
+	return c.BasicCluster
+}
+
+// GetSharedConfig returns the shared config.
+func (c *Cluster) GetSharedConfig() sc.SharedConfigProvider {
+	return c.persistConfig
+}
+
+// GetRuleManager returns the rule manager.
+func (c *Cluster) GetRuleManager() *placement.RuleManager {
+	return c.ruleManager
+}
+
+// GetRegionLabeler returns the region labeler.
+func (c *Cluster) GetRegionLabeler() *labeler.RegionLabeler {
+	return c.labelerManager
+}
+
+// GetStoresLoads returns load stats of all stores.
+func (c *Cluster) GetStoresLoads() map[uint64][]float64 {
+	return c.hotStat.GetStoresLoads()
+}
+
+// IsRegionHot checks if a region is in hot state.
+func (c *Cluster) IsRegionHot(region *core.RegionInfo) bool {
+	return c.hotStat.IsRegionHot(region, c.persistConfig.GetHotRegionCacheHitsThreshold())
+}
+
+// GetHotPeerStat returns hot peer stat with specified regionID and storeID.
+func (c *Cluster) GetHotPeerStat(rw utils.RWType, regionID, storeID uint64) *statistics.HotPeerStat {
+	return c.hotStat.GetHotPeerStat(rw, regionID, storeID)
+}
+
+// RegionReadStats returns hot region's read stats.
+// The result only includes peers that are hot enough.
+// RegionStats is a thread-safe method
+func (c *Cluster) RegionReadStats() map[uint64][]*statistics.HotPeerStat {
+	// As read stats are reported by store heartbeat, the threshold needs to be adjusted.
+	threshold := c.persistConfig.GetHotRegionCacheHitsThreshold() *
+		(utils.RegionHeartBeatReportInterval / utils.StoreHeartBeatReportInterval)
+	return c.hotStat.RegionStats(utils.Read, threshold)
+}
+
+// RegionWriteStats returns hot region's write stats.
+// The result only includes peers that are hot enough.
+func (c *Cluster) RegionWriteStats() map[uint64][]*statistics.HotPeerStat {
+	// RegionStats is a thread-safe method
+	return c.hotStat.RegionStats(utils.Write, c.persistConfig.GetHotRegionCacheHitsThreshold())
+}
+
+// BucketsStats returns hot region's buckets stats.
+func (c *Cluster) BucketsStats(degree int, regionIDs ...uint64) map[uint64][]*buckets.BucketStat {
+	return c.hotStat.BucketsStats(degree, regionIDs...)
+}
+
+// GetPersistOptions returns the persist options.
+func (c *Cluster) GetPersistOptions() sc.ConfProvider {
+	return c.persistConfig
+}
```

Comment:
This is kind of weird to me since the method name is `GetPersistOptions` but returns an interface. Maybe we could keep it returning `config.PersistOptions` and give it to the caller to choose the specific acceptance type?

Reply:
The function is part of the `ClusterInformer` interface.

Reply:
And I remove the GetPersistOptions from that interface.